test_runner/src/main/kotlin/ftl/args/IosArgs.kt

```diff
@@ -69,28 +70,26 @@ private fun IosArgs.calculateShardChunks() = if (disableSharding)
     emptyList() else
     ArgsHelper.calculateShards(
         filteredTests = filterTests(findTestNames(xctestrunFile), testTargets)
-            .distinct()
-            .map { FlankTestMethod(it, ignored = false) },
+                .flatMap { it.value }
+                .distinct()
+                .map { FlankTestMethod(it, ignored = false) },
         args = this
     ).shardChunks
 
 @VisibleForTesting
-internal fun filterTests(validTestMethods: List<String>, testTargetsRgx: List<String?>): List<String> {
-    if (testTargetsRgx.isEmpty()) {
-        return validTestMethods
-    }
-
-    return validTestMethods.filter { test ->
-        testTargetsRgx.filterNotNull().forEach { target ->
-            try {
-                if (test.matches(target.toRegex())) {
-                    return@filter true
+internal fun filterTests(
+    validTestMethods: XctestrunMethods,
+    testTargetsRgx: List<String>
+): XctestrunMethods =
+    if (testTargetsRgx.isEmpty()) validTestMethods
+    else validTestMethods.mapValues { (_, tests) ->
+        tests.filter { test ->
+            testTargetsRgx.any { target ->
+                try {
+                    test.matches(target.toRegex())
```

Comment:
Let's map `testTargetsRgx` into regex (its name actually indicates to be a list of regex) and reuse it. Current implementation parse all `testTargetsRgx` members for each `validTestMethods` value separately